### PR TITLE
Improve Wagtail page statuses to be more descriptive

### PR DESCRIPTION
**Pull request by kurtw**
This is going to be a pain to review. Sorry in advance. It should make you feel a bit less worried about our shared vs live vs draft state handling. Anyway, with the addition of `sharing` pages in Wagtail, we've complicated the statuses. There are certain odd states of the pages that required more descriptive statuses so this fixes that. More in the notes.

## Additions

- `has_unshared_changes` field to CFGOVPage

## Changes

- `CFGOVPage.serve()` now checks the status of the page's `has_unshared_changes` field before serving the page.
- Our hook function `share_the_page()` sets state of `has_unshared_changes` appropriately.

## Testing

- Run `./cfgov/manage.py migrate`
- Take a page all the way down to the `DRAFT` state.
 - Check that the status is `DRAFT`
- `SHARE` it.
 - Check that the status of the page is `SHARED`
 - Check that it exists as it's supposed to.
 - Check that it doesn't exist on the live site.
- Make an edit. The hit `SAVE AS DRAFT`.
 - Check that the status of the page is `SHARED + DRAFT`
 - Check that the edit didn't show on the shared site.
 - Check that the page still doesn't exist on the live site.
- Publish the page.
 - Check that the status of the page is `LIVE`
 - Check that the page exists on both sites.
- Make edits and save a draft
 - Check that the status of the page is `LIVE + DRAFT`
 - Make sure edits don't end up on either site
- Share the page.
 - Check that the status of the page is `LIVE + SHARED`
 - Verify the edits now show on the shared site but not production.
- Save a draft.
 - Check that the status of the page is `LIVE + (SHARED + DRAFT)`
- Unpublish
 - Check that the status of the page is `SHARED + DRAFT`
 - Verify the production site page is not longer accessible.
- Unshare
 - Check that the status of the page is `DRAFT`
 - Verify the shared site page is no longer accessible

## Review

- @kave 
- @richaagarwal 
- @rosskarchner 

## Notes

**New page status reference**
1. The page is just started, it is neither SHARED nor LIVE. — DRAFT
2. The page moves from draft to share on content but is not LIVE — SHARED
3. The page is SHARED and has unshared updates — SHARED + DRAFT
4. The page is published to www (and shared on content) -- LIVE
5. The page is LIVE and has unpublished updates -- LIVE + DRAFT
6. The page is LIVE and has shared updates -- LIVE + SHARED
7. The page is LIVE and has shared updates and has unpublished updates - LIVE + (SHARED+DRAFT)

## Todos

- Document this later on before we're training folks on Wagtail

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by richaagarwal** [comment]
@kurtw 

"Make edits and save a draft
Check that the status of the page is LIVE + DRAFT
Make sure edits don't end up on either site"  - This shows up as LIVE + (SHARED + DRAFT) for me. 

And relatedly, I'm not sure I understand the difference between these two:

5. The page is LIVE and has unpublished updates -- LIVE + DRAFT
7. The page is LIVE and has shared updates and has unpublished updates - LIVE + (SHARED+DRAFT)

In both cases, the page is live and shared, and if you modify any content, it's in draft for both of them. If you share the page, then the status ends up "LIVE + SHARED".  

In short, I'm unable to create LIVE + DRAFT, only LIVE + (SHARED + DRAFT). 

But, maybe we should use "LIVE + DRAFT" for when a page has been published, not shared, and also has edits that have been published. Right now, that shows as "DRAFT" which is kind of misleading.

**Comment by schaferjh** [comment]
@richaagarwal The intended difference in scenario form:

LIVE + DRAFT: The page is live, and meanwhile I made edits to it in Wagtail that are saved, but not pushed to the content server (so not shared) and not on the live version (not published)

LIVE + (SHARED + DRAFT): The page is live, meanwhile discussions are happening about updates -- an updated version has been shared to the content server for stakeholder review, meanwhile I have started to make additional edits to it in Wagtail (perhaps some stakeholder feedback has come through but not all) and saved those edits but not shared them to the content server.

Does that make sense?

**Comment by richaagarwal** [comment]
@schaferjh Yes, that makes sense, but that's not what's happening :)

**Comment by kurtw** [comment]
Closing to do more testing.

**Comment by kurtw** [comment]
Ready for review again.

**Comment by kurtw** [comment]
This should be addressed at a later date.
